pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -88,6 +90,15 @@ func NewGlobalIngressIPController(config syncer.ResourceSyncerConfig, pool *ipam
 					return err
 				}
 
+				key, _ := cache.MetaNamespaceKeyFunc(obj)
+				return controller.iptIface.AddEgressRulesForHeadlessSVCPods(key, target, reservedIPs[0], globalNetIPTableMark)
+			} else if gip.Spec.Target == submarinerv1.HeadlessServiceEndpoints {
+				// TODO: fixme (Maybe only rename and change messages?)
+				err := controller.iptIface.AddIngressRulesForHeadlessSvcPod(reservedIPs[0], target)
+				if err != nil {
+					return err
+				}
+
```

Comment:
This is same as L88-L94. I agree, we can rename it and match both Target with submarinerv1.HeadlessServicePod || submarinerv1.HeadlessServiceEndpoints